### DISCUSSION TITLE
Webhook test delivery

### DISCUSSION THREAD

**Discussion by almereyda** [outline/outline#6484] [Ideas]
> As a platform integrator, I need to send test Webhook events, in order to validate the receiving side.

Currently trying to work around the missing Mattermost integration #4271 by building a small filter pipeline, which converts the Outline webhook JSON to Mattermost webhook JSON with a workflow automation tool, in our case n8n, but could be any.

While it has been stated earlier that Webhook features are currently on hold

- https://github.com/outline/outline/discussions/6318#discussioncomment-7961203

we still find the implementation in an early stage. If not for a Test button, there was at least the request for a view of recent Webhook deliveries.

* https://github.com/outline/outline/issues/3702
* https://github.com/outline/outline/pull/3833

While that may not be a sufficient condition to send test deliveries, the two features could seemingly integrate well.

**Comment by tommoor** [comment]
The API's are there for all this stuff, we're already recording deliveries in the db … as you saw we're just not prioritizing changes here as the truth is barely anybody uses webhooks so there is much higher value things to work on with the limited time available.

As with the #6318 – happy to accept contributions in this arena

**Comment by almereyda** [comment]
Glad to hear. I'm slowly working in the direction, but no ETA. A small button to initiate a new `test` event shouldn't be too hard to implement and wire up for a dedicated mind.

A very minimmal n8n workflow to convert Outline webhooks into Mattermost webhooks has been published in

- https://github.com/outline/outline/issues/4271#issuecomment-1925466610

**Reply by almereyda** [reply]
There's more, and I think we're getting there by dogfooding our integrations.

- https://github.com/outline/outline/issues/6451
- https://github.com/outline/outline/discussions/6487
